### PR TITLE
[Gluten-825]Enable compiling in macOS.

### PR DESCRIPTION
**Pull request by Yohahaha**
## What changes were proposed in this pull request?

Platform:
MacOS 12.3 (Intel)
Apple clang 13.1.6
Homebrew 3.6.17

This pr enabling compile Gluten in macOS without Docker,  pr #819 do similar things with Docker.
Add some no-error compile options, and modify some code to get macOS compatibility.
#825 

References:
[CPU Affinity on macOS](https://github.com/facebookarchive/caffe2/issues/306)
[GTest with CMake](https://google.github.io/googletest/quickstart-cmake.html)
[-Wpessimizing-move](https://stackoverflow.com/questions/62061433/how-to-avoid-the-pessimizing-move-warning-of-nrvo)

## How was this patch tested?

manul test.

Contract me if any problem in your macOS.

### CODE REVIEW

**Comment by Yohahaha** [comment]
cc @CodingCat @PHILO-HE

**Comment by FelixYBW** [comment]
@CodingCat you may try this PR

File: cpp/core/jni/JniWrapper.cc [hunk 1]
```diff
@@ -908,18 +907,6 @@ Java_io_glutenproject_memory_alloc_NativeMemoryAllocator_bytesAllocated(JNIEnv*
   JNI_METHOD_END(-1L)
 }
 
-JNIEXPORT void JNICALL Java_io_glutenproject_tpc_MallocUtils_mallocTrim(JNIEnv* env, jobject obj) {
```
Comment (Yohahaha) [inline cpp/core/jni/JniWrapper.cc:911]: After search all project about this interface, I can not find any usage.

Comment (FelixYBW) [reply cpp/core/jni/JniWrapper.cc:911]: @zhztheplayer Can we remove the API?

**Comment by CodingCat** [comment]
can we set a CI/CD step to build on mac? just to ensure that what is committed in this PR will not break

**Comment by CodingCat** [comment]
also as discussed in slack, build_velox.sh also need some non-tiny changes....(is the correctness of the script covered by the current CI/CD?)

**Comment by CodingCat** [comment]
I tested this PR, it didn't work in my local, seems GTEST changes is the deal breaker

**Comment by Yohahaha** [comment]
After thinking about real usage, build and run in macOS without docker has less benefits, so i would close this pr.
#819 should be a better way.

File: cpp/core/jni/JniWrapper.cc (hunk 1 above)
Comment (zhztheplayer) [reply cpp/core/jni/JniWrapper.cc:911]: +1, it should be removed.

**Pull request closed without merging**